mikeio1d/res1d.py

```diff
@@ -130,15 +137,72 @@ def __repr__(self):
 
     # region Private methods
 
-    def _get_actual_queries(self, queries):
-        """Finds out which list of queries should be used."""
-        queries = self._queries if queries is None else queries
-        queries = queries if isinstance(queries, list) else [queries]
-        return queries
+    def _get_timeseries_ids_to_read(
+        self, queries: List[QueryData] | List[TimeSeriesId]
+    ) -> List[TimeSeriesId]:
+        """Find out which list of TimeSeriesId objects should be used for reading.
+
+        If user supplies queries, then convert them to TimeSeriesId. Otherwise use the
+        current queue of TimeSeriesId objects.
+
+        Parameters
+        ----------
+        queries : List[QueryData] | List[TimeSeriesId]
+            List of queries or time series ids supplied in read() method.
+
+        Returns
+        -------
+        List of TimeSeriesId objects.
+        """
+        if queries is None:
+            return self.result_network.queue
+
+        timeseries_ids = self._validate_queries_as_timeseries_ids(queries)
+
+        return timeseries_ids
+
+    def _validate_queries_as_timeseries_ids(
```

Comment:
I suggest to move the contents of the method to result_query module. I think the idea with the Res1D class, that it is collecting high level method call and has as little as possible particularities.